Source/Extensions/Blazorise.Components/DropdownList.razor.cs

```diff
@@ -103,6 +105,34 @@ private bool GetItemDisabled( TItem item )
         return DisabledItem.Invoke( item );
     }
 
+    private void FilterData( IQueryable<TItem> query )
+    {
+        dirtyFilter = false;
+        if ( !FilterEnabled || string.IsNullOrEmpty( FilterText ) )
+        {
+            filteredData = Data;
+            return;
+        }
+        
+        if ( query == null )
+        {
+            filteredData = Enumerable.Empty<TItem>();
+            return;
+        }
+
+        if ( TextField == null )
+            return;
+
+        filteredData = Data.Where( x => TextField.Invoke( x ).Contains( FilterText, StringComparison.OrdinalIgnoreCase ) );
```

Comment:
Do we need to store filtered data? Or can we just return it as 

`return Data.Where( x => TextField.Invoke( x ).Contains( FilterText, StringComparison.OrdinalIgnoreCase ) );`

Reply:
Yeah, in this the `filteredData` field isn't needed. That implementation can go without it.

I took it form the `Autocomplete`, where it is an `IList` and the data aren't enumerated on every get. But made it `IEnumberable`, because the Data are also `IEnumerable`.

Well - it is a question now. Should the `DropdownList.FilteredData` also be an `IList` instead of `IEnumerable`?

- IEnumerable will do the filtering on every enumeration. Possibly on every re-render. This could have significant perf implications.
- If converted to List, the filtering will be done only on text change, but it needs to fit into memory.... This is also a concern for current implementation of `Autocomplete`...

Reply:
Yeah, performance can be a problem. Dropdown, from a UX perspective is meant to be used only for small menus. With this new search features we are essentially breaking that "behavior" since we are allowing for very large menus. I guess that is fine if our users would expect that. But now we have a dilemma. Should we optimize or leave it as it is.

I think we can leave it as is, and optimize with caching later if it becomes a problem.